### DISCUSSION TITLE
budget constraint for Optimal Transport

### DISCUSSION THREAD

**Discussion by jihwanp** [PythonOT/POT#426] [Q&A]
Hi im new to optimal transport problem and happy to find such a repo that provides many variants of optimal transport algorithm.

It seems that most of the constraint that this repo solves for optimal transport problem assumes to have two marginal distribution. However in my case for the optimal transport problem, I only have a single marginal distribution, another constraint is just for the total summation of transport plan (budget). (equation below)
![image](https://user-images.githubusercontent.com/70801434/210161343-6d47d171-9bdd-496e-9ea1-f9d5f6dfb9d8.png)

Actually well-known LP solver like Gurobi can be used but I want to implement with some well-known nice algorithm for my case (should be faster), so I wonder if there is a tool that can be applied within POT or if there is an existing algorithm that can be used quickly in my case.

Thanks

**Comment by rflamary** [comment]
From a look at the optimization problem I dont see how it can be feasible exept when K=n. Indeed the first constraint forces each columns to sum to 1 so a double sum will be n no? You might want to look toward variants of unbalanced OT in order to not have constraints but penalizations.

Note that if K=n then the problem is called semi relaxed OT and a simple solution can be found (basocaly pi has values 1 on the argmin of C for each columns).
